nilearn/plotting/surf_plotting.py

```diff
@@ -75,10 +77,15 @@ def _ball_sampling(images, nodes, affine=None, ball_radius=3, n_points=20):
     sample_slices = [slice(None, None, None)] + np.s_[[
         ax + pad_size for ax in np.rollaxis(sample_indices, -1)
     ]]
-    samples = np.pad(
-        images, pad, mode='constant',
-        constant_values=np.nan)[sample_slices]
-    texture = np.nanmean(samples, axis=2)
+    try:
+        samples = np.pad(
+            images, pad, mode='constant',
+            constant_values=np.nan)[sample_slices]
+        texture = np.nanmean(samples, axis=2)
+        if not(np.isfinite(texture).all()):
+            raise IndexError()
```

Comment:
Perhaps consider adding a more verbose error message

Reply:
+1. This will not help the user.